fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -886,6 +916,11 @@ ACTOR Future<Void> preresolutionProcessing(CommitBatchContext* self) {
 	if (SERVER_KNOBS->ENABLE_VERSION_VECTOR_TLOG_UNICAST) {
 		self->writtenTagsPreResolution = self->getWrittenTagsPreResolution();
 	}
+
+	if (SERVER_KNOBS->SHARD_THROTTLING_ENABLED) {
+		self->throttledShard = self->checkHotShards();
```

Comment:
`throttledShard` should be set when Ratekeeper notifies CP, and cleared after expiration time.

If there are shards to be throttled, then we check the transaction batch. Mark overlapping txns as "shard_throttling_error" so that we can skip resolution for these transactions and can reply the error in the `reply()` call.

Reply:
> throttledShard should be set when Ratekeeper notifies CP, 

yes

> and cleared after expiration time

The ratekeeper could send a second ThrottleShards request for a different shard before the first request expired,  so the bookeeping of expiration times is a little more complex. (I have not written that code yet.)